client/post-editor/post-editor.jsx

```diff
@@ -650,14 +649,12 @@ export const PostEditor = React.createClass( {
 
 	onPublish: function() {
 		const edits = {
-			...this.props.edits,
-			status: 'publish'
+			status: 'publish',
+			...this.props.edits
 		};
 
-		// determine if this is a private publish
-		if ( utils.isPrivate( this.state.post ) ) {
-			edits.status = 'private';
-		} else if ( utils.isFutureDated( this.state.post ) ) {
+		// determine if this is a future publish
+		if ( utils.isFutureDated( this.state.post ) ) {
```

Comment:
Because effectively this changes logic from:

Before: We don't trust `edits`, apply `publish` status and decide after if we need `private` or `future`
After: We trust `edits`, but use `publish` as a default

With "After" case, we could do similar with future dating to apply it as an edit before calling `onPublish`.